### DISCUSSION TITLE
[Closed] Behaviors extension to VisualElementExtensions

### DISCUSSION THREAD

**Discussion by bijington** [CommunityToolkit/Maui.Markup#28] [New Feature Discussions]
I would like to propose that we add a `WithBehavior(this VisualElement visualElement, Behavior behavior)` extension that will add the supplied `behavior` to `visualElement.Behaviors`.

It would then allow us to simplify this:

```csharp
var entry = new Entry
{
    Placeholder = "Start typing until MaxLength is reached...",
    MaxLength = 100
};

entry.Behaviors.Add(
    new MaxLengthReachedBehavior().Bind(
        MaxLengthReachedBehavior.CommandProperty,
        nameof(ViewModel.MaxLengthReachedCommand)));

Content = entry;
```

down to this:

```csharp
Content = new Entry
{
    Placeholder = "Start typing until MaxLength is reached...",
    MaxLength = 100
}.WithBehavior(new MaxLengthReachedBehavior().Bind(
        MaxLengthReachedBehavior.CommandProperty,
        nameof(ViewModel.MaxLengthReachedCommand));
```

Or better yet as @pictos has suggested we could make the method accept `params` and the developer could hook up multiple behaviors (e.g. `WithBehaviors(params Behavior[] behaviors)`)

**Comment by pictos** [comment]
We can have both, that way if the dev just wants to add one behavior they will not pay for the array allocation.

**Comment by TheCodeTraveler** [comment]
Thanks Shaun + Pedro! I think this is ready to be promoted to a Proposal!

**Reply by bijington** [reply]
Done: https://github.com/CommunityToolkit/Maui.Markup/issues/29

**Comment by CliffAgius** [comment]
Closed and answered
